src/client.rs

```diff
@@ -72,19 +77,24 @@ impl Client {
         }
 
         let con = self.get_async_connection().await?;
-        crate::aio::SharedConnection::new(con, TokioExecutor)
+        crate::aio::MultiplexedConnection::new(con, TokioExecutor)
     }
 
-    /// Returns a async shared connection with a specific executor.
-    pub async fn get_shared_async_connection_with_executor<E>(
+    /// Returns an async multiplexed connection from the client.
+    ///
+    /// A multiplexed connection can be cloned, allowing requests to be be sent concurrently
+    /// on the same underlying connection (tcp/unix socket).
+    ///
+    /// Requires an executor to spawn a task that drives the underlying connection.
```

Comment:
I wonder if a note should be added that `get_multiplexed_async_connection` is also available, but only if the `executor` feature is enabled. Might be a bit hard to discover otherwise.

Reply:
Changed it around a bit to be `tokio-executor` instead and mentioned the feature.